### ISSUE TITLE
Run OSX tests on Travis-CI

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8592]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)| @adiroiban reported|
|-|-|
|Trac ID|trac#8592|
|Type|enhancement|
|Created|2016-07-12 14:44:43Z|
|Branch|https://github.com/twisted/twisted/tree/8592-travis-osx|

steal from the best 

https://github.com/pyca/cryptography/blob/master/.travis.yml

<details><summary>Searchable metadata</summary>

```
trac-id__8592 8592
type__enhancement enhancement
reporter__adiroiban adiroiban
priority__normal normal
milestone__None None
branch__8592_travis_osx 8592-travis-osx
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1468334683621636 1468334683621636
changetime__1469033265775542 1469033265775542
version__None None
owner__adiroiban adiroiban

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_1"></a>|@adiroiban commented|
|-|-|

https://github.com/twisted/twisted/pull/319

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_2"></a>|@adiroiban removed owner|
|-|-|

See the PR https://github.com/twisted/twisted/pull/319

I have pushed an initial version with minimal install and run scripts.

The scripts can be later expanded for pypy or for other python versions not supported by travis but provided by pyenv.

The Travis OSX build have been very slow recently (waiting for more than 2 hours) ... so as I am writing the last commit is still waiting to be executed on OSX

Some OSX tests are failing, Amber might know why :)  ... this is why OSX is now introduces as an unsupported travis job.

I think that fixing those tests should be the target of another ticket.

Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_3"></a>|@rodrigc set owner to @adiroiban|
|-|-|

This branch has conflicts

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_4"></a>|@adiroiban removed owner|
|-|-|

Thanks for the review!

Conflicts solved.... for now :)

I am working on various build system related stuff so there will always be conflicts

Please review latest changes. Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_5"></a>|@hawkowl set owner to @adiroiban|
|-|-|

Fix the one issue I commented with and merge.

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_6"></a>|GitHub set status to `closed`|
|-|-|

In changeset ba2ae0488b69d4ab85f232aa642d4120c73f49ac
```
#!CommitTicketReference repository="" revision="ba2ae0488b69d4ab85f232aa642d4120c73f49ac"
Merge pull request #319 from twisted/8592-travis-osx

Author: adiroiban
Reviewer: hawkowl
Fixes: #8592
```
